app/javascript/src/component_expander.js

```diff
@@ -34,6 +36,21 @@
 *       Lorem ipsum.....
 *     </dd>
 *   </dt>
+*
+*   Configuration:
+*   The config object can have the follwoing properties:
+*   - title: (jQuery|string)   If a jQuery object it should be the first child of
+*                              the $node. If the element itself is not a button, 
+*                              the element will be enhanced by wrapping its contents 
+*                              with a <button>.
+*                              If a string is provided, a button will be created and
+*                              prepended to the $node using the title text as the
+*                              button label.
```

Comment:
Similar to the title/activator config property.  These were both specific decisions

I deliberately backed away from handling multiple expanders within one element as going a step too far from what we already had...  An accordion component while functionally simlar would require extra work anyway (ensuring only one open at a time, expand all/contract all)

By not having the structure requirement, the code would get even more confusing as we have to do different things depending on whether we've been given an element (which could already be the activator) or a string (to enhance).  We'd also have to do a lot more work to figure out what is content and what is activator and how to wrap it if we wrap it etc..

Ultimately if I change the `title` config prop to `activator` and remove the structure expectation we will end up with the messy, unfocused trying-to-do-too-much component that I was attempting to avoid by refactoring this to a component that was focused on taking some specific markup and enhancing it into a disclosure widget.

I think the code gets worse the more 'generic' this 'component' tries to be.  Being opinionated about the markup allows it to remain simple and focused on a specific use case.